go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -324,7 +324,9 @@ func (ddb *DoltDB) ReadRootValue(ctx context.Context, h hash.Hash) (*RootValue,
 
 	if val != nil {
 		if rootSt, ok := val.(types.Struct); ok {
-			return &RootValue{ddb.db, rootSt}, nil
+			if rootSt.Name() == ddbRootStructName {
```

Comment:
Might combine this with prev if statement

Reply:
done